internal/alloydb/refresh.go

```diff
@@ -177,6 +177,9 @@ func fetchEphemeralCert(
 			err,
 		)
 	}
+	// Save the parsed certificate as the leaf certificate, to avoid additional
+	// parsing costs as part of the TLS connection.
+	cert.Leaf = clientCert
```

Comment:
does purely setting this reduce the parsing costs? or do we need to update somewhere else to access it?

Reply:
This is it. Since we're parsing the certificate above we just need to set this. For now, we're already depending on this to be set correctly to ensure we can read the expiration in dialer.go. I have a refactor PR that will remove the tight coupling between this code and dialer that is coming up next.